tests/emukit/benchmarking/test_benchmark_result.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import pytest
+from numpy.testing import assert_array_equal
+
+from emukit.benchmarking.loop_benchmarking.benchmark_result import BenchmarkResult
+
+
+def test_benchmark_result_initialization():
+    br = BenchmarkResult(['loop1', 'loop2'], 5, ['mae', 'mse'])
+
+    mae_data = br.extract_metric_as_array('loop1', 'mae')
+
+    assert mae_data.size == 0
+    assert mae_data.shape == (5, 0)
+
+    mse_data = br.extract_metric_as_array('loop1', 'mse')
+
+    assert mse_data.size == 0
+    assert mse_data.shape == (5, 0)
+
+
+def test_benchmark_result_invalid_names():
+    br = BenchmarkResult(['loop1', 'loop2'], 5, ['mae', 'mse'])
+
+    with pytest.raises(KeyError):
+        br.extract_metric_as_array('invalid loop', 'mse')
+
+    with pytest.raises(KeyError):
+        br.extract_metric_as_array('loop1', 'invalid metric')
+
+
+def test_benchmark_result_add_data():
+    br = BenchmarkResult(['loop1'], 2, ['mae', 'mse'])
+
+    br.add_results('loop1', 0, 'mae', np.array([1, 2, 3]))
+    br.add_results('loop1', 0, 'mse', np.array([1, 2]))
+    br.add_results('loop1', 1, 'mae', np.array([4, 5, 6]))
+    br.add_results('loop1', 1, 'mse', np.array([1, 2]))
```

Comment:
is there a reason you chose same numbers as in iter 0 for mse? Might be better to choose different ones.

Reply:
sorry, not very clear on my part. I meant that loop1-mse pair gets the same numbers on repeat 0 and 1. I like tests with differing numbers better.

Reply:
Oh, i see. That was actually intentional, to give same numbers to one pair and different numbers to other pair. For a slight chance that this will somehow impact the output. And the test verifies that it does not